cargo-quickbuild/src/description.rs

```diff
@@ -0,0 +1,84 @@
+use std::collections::BTreeSet;
+use std::fmt::Write as _;
+
+use crypto_hash::hex_digest;
+use crypto_hash::Algorithm;
+
+use cargo::core::PackageId;
+
+use crate::quick_resolve::QuickResolve;
+
+/// A self-contained description of a package build configuration
+pub struct PackageDescription {
```

Comment:
interestingly, PackageDescription has enough information to write down a Cargo.toml in a tempdir and spin up a whole new instance of cargo-quickbuild with its own `resolve`. This means that PackageDescription can also be our network request format.